### PR TITLE
[CORE-12562] Support React 18

### PR DESCRIPTION
**Pull request by spossner**
[JIRA Story](https://jira.nebenan.de/browse/CORE-12562)

Upgrade to React 18, drop mocha because no tests in the package (or keep for consistency?) and upgraded to storybook 7

### CODE REVIEW

File: src/components/stories/Component.jsx [hunk 1]
```diff
@@ -1,9 +1,6 @@
-/* eslint no-alert: "off" */
-import T from './t';
+import T from '../t';
 
-export default { title: 'T' };
-
-export const t = () => (
+export const Component = () => (
```
Comment (eliasbraun) [inline src/components/stories/Component.jsx]: Component might be a too generic name... 
Why did you switch from  T?

File: package.json [hunk 2]
```diff
@@ -42,28 +41,31 @@
...
+    "eslint-config-nebenan": "^4.4.0",
     "prop-types": "^15.7.2",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
     "rollup": "^2.33.3",
     "rollup-plugin-delete": "^2.0.0",
     "rollup-plugin-peer-deps-external": "^2.2.4",
-    "typescript": "^4.1.2"
```
Comment (eliasbraun) [inline package.json:61]: I wonder why that was in there before 🤔

File: src/components/stories/Component.stories.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { Component } from './Component';
+
+// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
```
Comment (eliasbraun) [inline src/components/stories/Component.stories.js]: I guess we can get rid of that comment

File: src/components/stories/Component.stories.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+export default {
+  title: 'Component',
+  component: Component,
+};
+
+// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
+export const Primary = {
+};
```
Comment (eliasbraun) [inline src/components/stories/Component.stories.js]: Was that WIP?

File: src/components/stories/InternationalHook.stories.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
+export default {
+  title: 'InternationalHook',
+  component: InternationalHook,
+};
+
+// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
+export const Primary = {
+};
```
Comment (eliasbraun) [inline src/components/stories/InternationalHook.stories.js]: 🤔

File: src/components/stories/LocaleHook.stories.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { LocaleHook } from './LocaleHook';
+
+// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
```
Comment (eliasbraun) [inline src/components/stories/LocaleHook.stories.js]: Same like above

File: src/components/stories/LocaleHook.stories.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+export default {
+  title: 'LocaleHook',
+  component: LocaleHook,
+};
+
+// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
```
Comment (eliasbraun) [inline src/components/stories/LocaleHook.stories.js]: Same

File: src/components/stories/THook.stories.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import { THook } from './THook';
+
+// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
```
Comment (eliasbraun) [inline src/components/stories/THook.stories.js]: same

File: src/components/stories/THook.stories.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+export default {
+  title: 'THook',
+  component: THook,
+};
+
+// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
+export const Primary = {
+};
```
Comment (eliasbraun) [inline src/components/stories/THook.stories.js]: same

**Review by eliasbraun — CHANGES_REQUESTED** [review]

**Review by peterhass — APPROVED** [review]
nice stories!

**Review by eliasbraun — APPROVED** [review]

**Pull request merged**